Cannot execute a brand new application, Error instantiating bean of type  [io.micronaut.security.oauth2.routes.OauthRouteBuilder]

I've created a microservice application from scratch just to give a try, ran a jhipster registry instance using the docker-compose file generated and than executed ./gradlew to see the result.

 Unfortunately it gave me this exception during boot:

`
13:50:16.476 [main] ERROR io.micronaut.runtime.Micronaut - Error starting Micronaut server: Bean definition [com.mycompany.myapp.config.LoggingConfiguration] could not be loaded: Error instantiating bean of type  [io.micronaut.security.oauth2.routes.OauthRouteBuilder]
`

`
Path Taken: new RxNettyHttpClientRegistry(HttpClientConfiguration defaultHttpClientConfiguration,HttpClientFilterResolver httpClientFilterResolver,[LoadBalancerResolver loadBalancerResolver],NettyClientSslBuilder nettyClientSslBuilder,ThreadFactory threadFactory,MediaTypeCodecRegistry codecRegistry,EventLoopGroupRegistry eventLoopGroupRegistry,EventLoopGroupFactory eventLoopGroupFactory,BeanContext beanContext) --> new DefaultLoadBalancerResolver(BeanContext beanContext,[List serviceInstanceLists]) --> new EurekaServiceInstanceList([EurekaConfiguration configuration],InstanceConfiguration instanceConfiguration) --> new EurekaConfiguration(EurekaConnectionPoolConfiguration eurekaConnectionPoolConfiguration,ApplicationConfiguration applicationConfiguration,[EurekaRegistrationConfiguration eurekaRegistrationConfiguration]) --> new EurekaRegistrationConfiguration([EmbeddedServer embeddedServer],ApplicationConfiguration applicationConfiguration,DataCenterInfo dataCenterInfo,String appName,String hostname,Integer port,String ipAddr,String instanceId,Boolean preferIpAddress) --> new NettyHttpServer(NettyHttpServerConfiguration serverConfiguration,ApplicationContext applicationContext,[Router router],RequestArgumentSatisfier requestArgumentSatisfier,MediaTypeCodecRegistry mediaTypeCodecRegistry,NettyCustomizableResponseTypeHandlerRegistry customizableResponseTypeHandlerRegistry,StaticResourceResolver resourceResolver,Provider ioExecutor,ThreadFactory threadFactory,ExecutorSelector executorSelector,ServerSslBuilder serverSslBuilder,List outboundHandlers,EventLoopGroupFactory eventLoopGroupFactory,EventLoopGroupRegistry eventLoopGroupRegistry,HttpCompressionStrategy httpCompressionStrategy,HttpContentProcessorResolver httpContentProcessorResolver,ChannelOptionFactory channelOptionFactory) --> new DefaultRouter([Collection builders]) --> new OauthRouteBuilder(ExecutionHandleLocator executionHandleLocator,UriNamingStrategy uriNamingStrategy,ConversionService conversionService,BeanContext beanContext,OauthConfiguration oauthConfiguration,OauthRouteUrlBuilder oauthRouteUrlBuilder,[List controllerList])
io.micronaut.context.exceptions.BeanInstantiationException: Bean definition [com.mycompany.myapp.config.LoggingConfiguration] could not be loaded: Error instantiating bean of type  [io.micronaut.security.oauth2.routes.OauthRouteBuilder]
`

`
Path Taken: new RxNettyHttpClientRegistry(HttpClientConfiguration defaultHttpClientConfiguration,HttpClientFilterResolver httpClientFilterResolver,[LoadBalancerResolver loadBalancerResolver],NettyClientSslBuilder nettyClientSslBuilder,ThreadFactory threadFactory,MediaTypeCodecRegistry codecRegistry,EventLoopGroupRegistry eventLoopGroupRegistry,EventLoopGroupFactory eventLoopGroupFactory,BeanContext beanContext) --> new DefaultLoadBalancerResolver(BeanContext beanContext,[List serviceInstanceLists]) --> new EurekaServiceInstanceList([EurekaConfiguration configuration],InstanceConfiguration instanceConfiguration) --> new EurekaConfiguration(EurekaConnectionPoolConfiguration eurekaConnectionPoolConfiguration,ApplicationConfiguration applicationConfiguration,[EurekaRegistrationConfiguration eurekaRegistrationConfiguration]) --> new EurekaRegistrationConfiguration([EmbeddedServer embeddedServer],ApplicationConfiguration applicationConfiguration,DataCenterInfo dataCenterInfo,String appName,String hostname,Integer port,String ipAddr,String instanceId,Boolean preferIpAddress) --> new NettyHttpServer(NettyHttpServerConfiguration serverConfiguration,ApplicationContext applicationContext,[Router router],RequestArgumentSatisfier requestArgumentSatisfier,MediaTypeCodecRegistry mediaTypeCodecRegistry,NettyCustomizableResponseTypeHandlerRegistry customizableResponseTypeHandlerRegistry,StaticResourceResolver resourceResolver,Provider ioExecutor,ThreadFactory threadFactory,ExecutorSelector executorSelector,ServerSslBuilder serverSslBuilder,List outboundHandlers,EventLoopGroupFactory eventLoopGroupFactory,EventLoopGroupRegistry eventLoopGroupRegistry,HttpCompressionStrategy httpCompressionStrategy,HttpContentProcessorResolver httpContentProcessorResolver,ChannelOptionFactory channelOptionFactory) --> new DefaultRouter([Collection builders]) --> new OauthRouteBuilder(ExecutionHandleLocator executionHandleLocator,UriNamingStrategy uriNamingStrategy,ConversionService conversionService,BeanContext beanContext,OauthConfiguration oauthConfiguration,OauthRouteUrlBuilder oauthRouteUrlBuilder,[List controllerList])
        at io.micronaut.context.DefaultBeanContext.initializeContext(DefaultBeanContext.java:1549)
        at io.micronaut.context.DefaultApplicationContext.initializeContext(DefaultApplicationContext.java:220)
        at io.micronaut.context.DefaultBeanContext.readAllBeanDefinitionClasses(DefaultBeanContext.java:2780)
        at io.micronaut.context.DefaultBeanContext.start(DefaultBeanContext.java:233)
        at io.micronaut.context.DefaultApplicationContext.start(DefaultApplicationContext.java:166)
        at io.micronaut.runtime.Micronaut.start(Micronaut.java:64)
        at com.mycompany.myapp.DefaultmicronautApp.main(DefaultmicronautApp.java:69)
Caused by: io.micronaut.context.exceptions.BeanInstantiationException: Error instantiating bean of type  [io.micronaut.security.oauth2.routes.OauthRouteBuilder]
`

`
Path Taken: new RxNettyHttpClientRegistry(HttpClientConfiguration defaultHttpClientConfiguration,HttpClientFilterResolver httpClientFilterResolver,[LoadBalancerResolver loadBalancerResolver],NettyClientSslBuilder nettyClientSslBuilder,ThreadFactory threadFactory,MediaTypeCodecRegistry codecRegistry,EventLoopGroupRegistry eventLoopGroupRegistry,EventLoopGroupFactory eventLoopGroupFactory,BeanContext beanContext) --> new DefaultLoadBalancerResolver(BeanContext beanContext,[List serviceInstanceLists]) --> new EurekaServiceInstanceList([EurekaConfiguration configuration],InstanceConfiguration instanceConfiguration) --> new EurekaConfiguration(EurekaConnectionPoolConfiguration eurekaConnectionPoolConfiguration,ApplicationConfiguration applicationConfiguration,[EurekaRegistrationConfiguration eurekaRegistrationConfiguration]) --> new EurekaRegistrationConfiguration([EmbeddedServer embeddedServer],ApplicationConfiguration applicationConfiguration,DataCenterInfo dataCenterInfo,String appName,String hostname,Integer port,String ipAddr,String instanceId,Boolean preferIpAddress) --> new NettyHttpServer(NettyHttpServerConfiguration serverConfiguration,ApplicationContext applicationContext,[Router router],RequestArgumentSatisfier requestArgumentSatisfier,MediaTypeCodecRegistry mediaTypeCodecRegistry,NettyCustomizableResponseTypeHandlerRegistry customizableResponseTypeHandlerRegistry,StaticResourceResolver resourceResolver,Provider ioExecutor,ThreadFactory threadFactory,ExecutorSelector executorSelector,ServerSslBuilder serverSslBuilder,List outboundHandlers,EventLoopGroupFactory eventLoopGroupFactory,EventLoopGroupRegistry eventLoopGroupRegistry,HttpCompressionStrategy httpCompressionStrategy,HttpContentProcessorResolver httpContentProcessorResolver,ChannelOptionFactory channelOptionFactory) --> new DefaultRouter([Collection builders]) --> new OauthRouteBuilder(ExecutionHandleLocator executionHandleLocator,UriNamingStrategy uriNamingStrategy,ConversionService conversionService,BeanContext beanContext,OauthConfiguration oauthConfiguration,OauthRouteUrlBuilder oauthRouteUrlBuilder,[List controllerList])
        at io.micronaut.context.DefaultBeanContext.doCreateBean(DefaultBeanContext.java:1925)
        at io.micronaut.context.DefaultBeanContext.addCandidateToList(DefaultBeanContext.java:2990)
        at io.micronaut.context.DefaultBeanContext.getBeansOfTypeInternal(DefaultBeanContext.java:2891)
        at io.micronaut.context.DefaultBeanContext.getBeansOfType(DefaultBeanContext.java:1045)
        at io.micronaut.context.AbstractBeanDefinition.lambda$getBeansOfTypeForConstructorArgument$9(AbstractBeanDefinition.java:1143)
        at io.micronaut.context.AbstractBeanDefinition.resolveBeanWithGenericsFromConstructorArgument(AbstractBeanDefinition.java:1820)
        at io.micronaut.context.AbstractBeanDefinition.getBeansOfTypeForConstructorArgument(AbstractBeanDefinition.java:1138)
        at io.micronaut.context.AbstractBeanDefinition.getBeanForConstructorArgument(AbstractBeanDefinition.java:990)
`

##### **Reproduce the error**
- Create a brand new app with default parameters (see configuration below)
- Run an instance of jhipster registry using `docker-compose -f src/main/docker/jhipster-registry.yml up` after set the OAuth2 issuer, client-id and client-secret
- Put the same parameter in `src/main/resources/application.yml`
- Run `./gradlew`

##### **Related issues**

Not found related issues

##### **Project configuration**

```
/home/vcupelloni/defaultmicronaut
└─┬ generator-jhipster-micronaut@0.7.0
  └── generator-jhipster@6.10.5 

```


##### **JHipster configuration, a `.yo-rc.json` file generated in the root folder**


<details>
<summary>.yo-rc.json file</summary>
<pre>
{
  "generator-jhipster": {
    "promptValues": {
      "packageName": "com.mycompany.myapp"
    },
    "jhipsterVersion": "6.10.5",
    "applicationType": "microservice",
    "baseName": "defaultmicronaut",
    "packageName": "com.mycompany.myapp",
    "packageFolder": "com/mycompany/myapp",
    "serverPort": "8081",
    "authenticationType": "oauth2",
    "cacheProvider": "ehcache",
    "enableHibernateCache": true,
    "websocket": false,
    "databaseType": "sql",
    "devDatabaseType": "h2Disk",
    "prodDatabaseType": "postgresql",
    "messageBroker": false,
    "serviceDiscoveryType": "eureka",
    "buildTool": "gradle",
    "jwtSecretKey": "YourJWTSecretKeyWasReplacedByThisMeaninglessTextByTheJHipsterInfoCommandForObviousSecurityReasons",
    "embeddableLaunchScript": false,
    "creationTimestamp": 1611232810219,
    "jhiPrefix": "jhi",
    "entitySuffix": "",
    "dtoSuffix": "DTO",
    "otherModules": [
      {
        "name": "generator-jhipster-micronaut",
        "version": "0.7.0"
      }
    ],
    "enableTranslation": false,
    "clientPackageManager": "npm",
    "blueprints": [
      {
        "name": "generator-jhipster-micronaut",
        "version": "0.7.0"
      }
    ],
    "skipClient": true,
    "skipUserManagement": true
  }
}

</pre>
</details>


##### **JDL for the Entity configuration(s) `entityName.json` files generated in the `.jhipster` directory**

<details>
<summary>JDL entity definitions</summary>

<pre>

</pre>
</details>


##### **Environment and Tools**

openjdk version "11.0.10" 2021-01-19
OpenJDK Runtime Environment (build 11.0.10+8)
OpenJDK 64-Bit Server VM (build 11.0.10+8, mixed mode)

git version 2.30.0

node: v10.23.1

npm: 6.14.11

yeoman: 3.1.1

yarn: 1.22.10

Docker version 20.10.2, build 2291f610ae

docker-compose version 1.28.0, build unknown

INFO! Congratulations, JHipster execution is complete!

##### **Browsers and Operating System**

Chromium on Arch Linux

-   [ X ] Checking this box is mandatory (this is just to show you read everything)

--- a/generators/constants.js
+++ b/generators/constants.js
@@ -43,7 +43,7 @@ module.exports = {
         },
     },
     versions: {
-        micronaut: '2.4.2',
+        micronaut: '2.4.4',
         micronautData: '2.3.1',
         micronautOpenApi: '2.3.1',
 

--- a/generators/server/templates/gradle.properties.ejs
+++ b/generators/server/templates/gradle.properties.ejs
@@ -66,7 +66,7 @@ jaxb_runtime_version=2.3.2
 <%_ } _%>
 
 # gradle plugin version
-micronaut_plugin_version=1.4.0
+micronaut_plugin_version=1.4.5
 git_properties_plugin_version=2.2.2
 <%_ if (!skipClient) { _%>
 gradle_node_plugin_version=2.2.4

